### PR TITLE
Past connection

### LINKED ISSUE #4: Create past connections List
As a user, I want to see a list of previous connections that I have used so I can quickly access a previous configuration. 

AC: 
- On click of connection, an attempt to connect is made
  - Display error if there is one
- Clicking "Connect to existing" goes to new page that shows full list of information including Name, Type, Address, User
- Truncated list on Home screen that just shows nickname, left column

Callouts:
- manage global state with React Contexts

### PR DESCRIPTION
**Pull request by Ruben-agredano**
# Description

Made table dynamic by calling backend for a list of Connections.
The list is limited to the 5 most recent connections.
with Aditya's help, all connections are set to manual.
Use procedureDropdown to create a new connection.

## Link to the issue you are closing

Fixes #4 

## Screenshot or other testing that you have completed

[Screen Recording 2022-09-30 at 1.21.43 AM.mov.zip](https://github.com/JID-2111/JID-2111/files/9680999/Screen.Recording.2022-09-30.at.1.21.43.AM.mov.zip)

Had to compress otherwise GitHub wouldn't let me upload a video

# Checklist:

- [x] ESLint passes
- [x] Prettier passes
- [x] Included comments where necessary
- [x] Updated README if needed
- [x] Verified that code runs and generates no errors/warnings

### CODE REVIEW

**Review by alex-hobby — APPROVED** [review]
This looks great - tested locally and was able to see an added connection + could connect to it by clicking on the name.

File: src/renderer/components/RecentConnections.tsx [hunk 1]
```diff
@@ -1,62 +1,58 @@
...
       <h1 className="Header">Recent Connections</h1>
       <div className="TDiv">
         <Table className="table">
           <thead>
             <tr>
-              <th>ID</th>
-              <th>Type of Connection</th>
+              <th>Nick Name</th>
+              <th>Database Type</th>
+              <th>Connection type</th>
```
Comment (adityavidyadharan) [inline src/renderer/components/RecentConnections.tsx:29]:
```suggestion
```

File: src/renderer/components/RecentConnections.tsx [hunk 2]
```diff
@@ -1,62 +1,58 @@
...
+                return (
+                  <tr key={value.id}>
+                    <td className="LinkTD">
+                      <Link to="/Execute">
+                        <button className="buttonSelect" type="button">
+                          {value.nickname}
+                        </button>
+                      </Link>
+                    </td>
+                    <td>{value.type}</td>
```
Comment (adityavidyadharan) [inline src/renderer/components/RecentConnections.tsx:47]:
```suggestion
```

**Review by adityavidyadharan — CHANGES_REQUESTED** [review]
UI and functionality look great.
I think that we can remove the type field since everything is now *manual* and we'll be removing that distinction in the future.

**Review by adityavidyadharan — APPROVED** [review]
PR looks good!

**Pull request merged**